src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy

```diff
@@ -847,9 +848,24 @@ class OpenShiftDSL implements Serializable {
         private ArrayList<String> objectList;
         private String invalidMessage;
 
-        public OpenShiftResourceSelector(String highLevelOperation, Object okind, Object qualifier) {
+        public OpenShiftResourceSelector(String highLevelOperation, Object okind_or_list, Object qualifier) {
             super(highLevelOperation);
-            String kind = toSingleString(okind);
+
+            if (okind_or_list instanceof List) {
+                objectList = toStringList(okind_or_list);
+                this.kind = null;
+                this.invalidMessage = null;
+                return;
+            }
+
+            if (okind_or_list instanceof Object[]) {
+                objectList = new ArrayList<String>(Arrays.asList(toStringArray(okind_or_list)));
+                this.kind = null;
+                this.invalidMessage = null;
+                return;
+            }
+
+            String kind = toSingleString(okind_or_list);
             kind = kind==null?"all":kind;
```

Comment:
Is this valid?

```
$ oc get all dc/jenkins route/jenkins
error: there is no need to specify a resource type as a separate argument when passing arguments in resource/name form (e.g. 'oc get resource/<resource_name>' instead of 'oc get resource resource/<resource_name>'
```

Reply:
Oh wait nvm the others return early